server/test/unit/server/controllers/test_units.py

```diff
@@ -57,3 +57,38 @@ def test_results(self):
         # turn into list so the generator will be evaluated
         result = list(units_controller.find_units(units_iterable))
         self.assertEqual(result, [model_2_defined])
+
+
+@patch('pulp.plugins.loader.api.get_unit_model_by_id', spec_set=True)
+@patch('pulp.plugins.types.database.type_definition', spec_set=True)
+class TestGetUnitKeyFieldsForType(unittest.TestCase):
+    def test_returns_from_model(self, mock_type_def, mock_get_model):
+        """
+        test when the requested type is a mongoengine model
+        """
+        mock_get_model.return_value = DemoModel
+        mock_type_def.return_value = None
+
+        ret = units_controller.get_unit_key_fields_for_type('demo_model')
+
+        self.assertEqual(ret, ('key_field',))
```

Comment:
maybe deduplicate the definition by using `self.assertEqual(ret, DemoModel.unit_key_fields)`